impl/src/main/java/org/jboss/weld/contexts/AbstractBoundContext.java

```diff
@@ -85,4 +92,25 @@ public boolean dissociate(S storage) {
             return false;
         }
     }
+
+    @Override
+    public <T> Collection<ContextualInstance<T>> getAllContextualInstances() {
+        Set<ContextualInstance<T>> result = new HashSet<>();
+        // for instance lazily initialized conversation scope may be active but have null here
+        BeanStore beanStore = getBeanStore();
+        if (beanStore != null) {
+            getBeanStore().iterator().forEachRemaining((BeanIdentifier beanId) -> {
+                result.add(getBeanStore().get(beanId));
+            });
+        }
+        return result;
+    }
+
+    @Override
+    public <T> void clearAndSet(Collection<ContextualInstance<T>> setOfInstances) {
```

Comment:
This needs to be synchronized in some way for multithreaded contexts.

Reply:
By that you mean session and conversation I suppose? And only the HTTP backed I presume, so we could override it there perhaps. Not sure how to sync it yet, but good point.